src/main/groovy/nebula/plugin/publishing/publications/ShadowJarPlugin.groovy

```diff
@@ -0,0 +1,27 @@
+package nebula.plugin.publishing.publications
+
+import groovy.transform.CompileDynamic
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.jvm.tasks.Jar
+
+@CompileDynamic
+class ShadowJarPlugin implements Plugin<Project> {
+
+    @Override
+    void apply(Project project) {
+        project.afterEvaluate {
+            project.plugins.withId('com.github.johnrengelman.shadow') {
+                boolean jarTaskEnabled = project.tasks.findByName('jar').enabled
+                if(!jarTaskEnabled) {
+                    project.configurations {
+                        [apiElements, runtimeElements].each {
+                            it.outgoing.artifacts.removeIf { it.buildDependencies.getDependencies(null).contains(project.tasks.named('jar', Jar).get()) }
```

Comment:
What is gradle take on this? I would kind of expect that if I disable `jar` task it would lead to its removal from outgoing configurations.

Reply:
It is actually the recommended way: https://github.com/gradle/gradle/pull/12107/files#diff-15ab88015354e8cc60dd1218119f2910R113